crates/ruff/src/rules/flake8_copyright/rules/copyright_header_absent.rs

```diff
@@ -0,0 +1,47 @@
+use ruff_diagnostics::Violation;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::newlines::Line;
+
+use crate::settings::Settings;
+
+use lazy_regex::Regex;
+#[violation]
+pub struct HeaderLacksCopyright;
+
+impl Violation for HeaderLacksCopyright {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Copyright notice not present")
+    }
+}
+/// ## What it does
+/// Checks for Copyright Header to exist within at the top of a file within `copyright_min_file_size chars`
+/// format Copyright (C) <year> <author>
+///
+/// Error code C801
+pub(crate) fn copyright_header_absent(
+    line: &Line,
+    settings: &Settings,
+    current_char_index: i64,
+) -> Option<bool> {
+    let copyright_regexp = format!(
+        "{} {}",
+        settings.flake8_copyright.copyright_regexp, settings.flake8_copyright.copyright_author
+    );
+    let regex = Regex::new(copyright_regexp.trim()).unwrap();
```

Comment:
We should gracefully handle the case where the `Regex` is invalid rather than just panicking. 

@charliermarsh is there a place where we could do this deserialization outside of the rule (maybe in the configuration to settings deserialization?)

Reply:
Made it default to known string if user specified regex passed in for now, but open to better suggestions